mne/time_frequency/spectrum.py

```diff
@@ -1368,6 +1415,61 @@ def average(self, method="mean"):
         return Spectrum(state, **defaults)
 
 
+@fill_doc
+class EpochsSpectrumArray(EpochsSpectrum):
+    """Data object for precomputed epoched spectral data (in NumPy array format).
+
+    Parameters
+    ----------
+    data : array, shape (n_epochs, n_channels, n_freqs)
+        The channels' power spectral density for each epoch.
+    %(info_not_none)s
+    %(freqs_tfr)s
+    %(events_epochs)s
+    %(event_id)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.create_info
+    mne.EpochsArray
+    SpectrumArray
+
+    Notes
+    -----
+        .. versionadded:: 1.5.0
+    """
+
+    @verbose
+    def __init__(
+        self,
+        data,
+        info,
+        freqs,
+        events=None,
+        event_id=None,
+        *,
+        verbose=None,
+    ):
+        self.__setstate__(
+            dict(
+                method="array",
+                data=data,
+                sfreq=info["sfreq"],
+                dims=("epoch", "channel", "freq"),
+                freqs=freqs,
+                inst_type_str="Array",
+                data_type="Unknown",
```

Comment:
OK I'm convinced by your argument @alexrockhill that we can infer `data_type` from the `dtype`... It will make the `__repr__` much nicer.  I don't think we should say "Fourier coefficients" or "(Averaged) Power spectrum" though; I'm thinking maybe "Complex spectrum" or "Real spectrum". I do think that `method` should be "unknown" though, so we would end up with

```
<Real spectrum (from Array, unknown method) | {dims}, {freq_range}>
<Complex spectrum (from Array, unknown method) | {dims}, {freq_range}>
```

Reply:
Sounds good, looks like `Power Spectrum` is used elsewhere so I put that. Reading your comment more closely, you don't think power spectrum and prefer real though. I just think that might disambiguate as a double check for anyone providing amplitude that in fact it is supposed to be power but I can change to `Real` if you prefer.

Reply:
> I don't think we should say "Fourier coefficients" or "(Averaged) Power spectrum"

I thought this was pretty clear, but since you didn't do this I'll spell out my reasoning: I don't think that seeing "power spectrum" in the repr is going to be effective at reminding people that they should only pass power (not amplitude) values to the constructor. I also don't see the point of (implicitly) restricting the input to be power, since there's effectively no way we can check/verify that it's power. I'd much rather make the documentation (including repr) be agnostic about what is being fed in, and make downstream methods capable of not doing the wrong thing (i.e., the plotting function should know when to put real units vs unknown/arbitrary units; users can always overwrite the plot axis labels if they know what form their data is).

By the way in local testing the repr raises an error; I'll push a commit to fix in a minute or two.

Reply:
That's fine with me to make it accept accept more general inputs i.e. amplitude, I just thought it was simpler to require people to use power since they can be easily interconverted and if that was the case to remind them in case they didn't see it in the constructor. I think that might be slightly easier to maintain although less general but again I'm not strongly opinionated about it, the real/complex is okay.

Reply:
I've fixed the REPR but after looking at what it would take to make the plotting functions agnostic to power vs amplitude I'm actually going to backpedal here; it's too much work.  I'll add a docstring note caveat that we assume power.